Add instructions for specifying image-name in CLI (#922)
astronomer/docs: astro/develop-project.md

This example assumes that the name of each of your Python packages is identical to the name of its corresponding GitHub repository. In other words,`mypackage1` is both the name of the package and the name of the repository.

#### Step 2: Update Dockerfile

> 1. In your Astro project, replace the contents of `Dockerfile` with the following code. Retain any previous build steps you may have already added so you can add them again later.

Reviewer: ```suggestion
```
Did you mean to delete this step? It looks similar to the next step and seems to reference the same code block.

> 1. In `Dockerfile`, add `AS stage` to the `FROM` line which specifies your Runtime image. For example, if you use Runtime 5.0.0, your `FROM` line would be:

Reviewer: ```suggestion
1. In `Dockerfile`, add `AS stage` to the `FROM` line which specifies your Runtime image. For example, if you use Runtime 5.0.0, your `FROM` line would be:
```

  :::

> 3. In `Dockerfile` after the `FROM` line specifying your Runtime image, add the following configuration:

Reviewer: ```suggestion
1. After the `FROM` line, add the following configuration:
```

  :::

> 4. Add any commands that you had in the original `Dockerfile`. These should be added in stage3 after the line `FROM stage1 AS stage3`.

Reviewer: ```suggestion
4. Optional. If you had any other commands in your original `Dockerfile`, add them after the line `FROM stage1 AS stage3`.
```

#### Step 3: Build a custom Docker image

> 1. We can first automatically generate a unique image name using the following command:

Reviewer: ```suggestion
1. Run the following command to automatically generate a unique image name:
```

    ```sh
    image_name=astro-$(date +%Y%m%d%H%M%S)
    ```

2. Run the following command to create a new Docker image from your `Dockerfile`. Replace `<ssh-key>` with your SSH private key file name.

    ```sh
    DOCKER_BUILDKIT=1 docker build -f Dockerfile --progress=plain --ssh=github="$HOME/.ssh/<ssh-key>" -t $image_name .
    ```

> 3. Optional. Test or deploy your DAGs. See [Build and Run a Project Locally](develop-project.md#build-and-run-a-project-locally) or [Deploy Code to Astro](deploy-code.md).

Reviewer: ```suggestion
3. Optional. Test your DAGs locally. See [Build and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).
```

> 4. Deploy the image using the astro-cli <TODO: Link to the astro deploy --image-name doc>.

Reviewer: ```suggestion
4. Deploy the image using the Astro CLI: 
```
Not sure this link is necessary given that all they really need to know about the flag/ usage is here.

:::

> #### Step 2: Update Dockerfile

Reviewer: Same comments here as in the previous topic

Author: Hopefully I hit them all

> 1. Replace the entire contents of `Dockerfile` using the following steps. Retain any previous build steps you may have already added so you can add them again later.

Reviewer: ```suggestion
1. Optional. Copy any build steps you already have in your `Dockerfile` and save them for later.
```

    - Add the environment variable `PIP_EXTRA_INDEX_URL` to instruct pip on where to look for non-public packages.
    - Install public and private Python-level packages from your `requirements.txt` file.

4. Add any commands that you had in the original `Dockerfile`. These should be added in stage3 after the line `FROM stage1 AS stage3`.

#### Step 3: Build a custom Docker image

> 1. We can first automatically generate a unique image name using the following command:

Reviewer 2: ```suggestion
1. Run the following command to automatically generate a unique image name:
```
